### PR TITLE
Add exportAs

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/src/core/annotations_impl/annotations.ts [hunk 1]
```diff
@@ -746,9 +746,38 @@ export class Directive extends Injectable {
    */
   hostInjector: List<any>;
 
+  _exportAs: string;
```
Comment (tbosch) [inline modules/angular2/src/core/annotations_impl/annotations.ts]: Why a getter? The other fields don't have getters...

Comment (tbosch) [inline modules/angular2/src/core/annotations_impl/annotations.ts]: Ah, just saw further down... 
But I think this logic needs to be duplicated in the transformers, if we put it here, as transformers don't use our annotation classes.

File: modules/angular2/src/core/annotations_impl/annotations.ts [hunk 2]
```diff
@@ -1006,6 +1039,32 @@ export class Component extends Directive {
...
+   * @View({
+   *   template: `<child-cmp #c="child"></child-cmp>`,
+   *   directives: [ChildCmp]
+   * })
+   * class MainComponent {
+   * }
+   *
+   * ```
+   */
+  get exportAs() { return isBlank(this._exportAs) ? "$implicit" : this._exportAs; }
```
Comment (tbosch) [inline modules/angular2/src/core/annotations_impl/annotations.ts]: See above, transformers don't use our annotation classes, so they need to duplicate this. Probably move it somewhere else...

File: modules/angular2/src/core/compiler/proto_view_factory.ts [hunk 3]
```diff
@@ -377,6 +369,40 @@ function _createElementBinder(protoView, boundElementIndex, renderElementBinder,
   return elBinder;
 }
 
+export function createDirectiveVariableBindings(renderElementBinder:renderApi.ElementBinder,
+                                         directiveBindings:List<DirectiveBinding>): Map<String, number> {
+  var directiveVariableBindings = MapWrapper.create();
```
Comment (tbosch) [inline modules/angular2/src/core/compiler/proto_view_factory.ts]: Wouldn't it be a little faster to reverse this:

1. loop over directives
2. for those that have a `exportAs`, look into the `renderElementBinder.variableBindings` (as the key is `exportAs`)

Comment (tbosch) [inline modules/angular2/src/core/compiler/proto_view_factory.ts]: Probably also a little simpler...

Comment (vsavkin) [inline modules/angular2/src/core/compiler/proto_view_factory.ts]: I was thinking about it, but it makes handling corner cases harder.

For example, we need to throw when no directives actually matched a variable binding. I think we have to loop over the map anyway.

**Pull request merged**